src/Raven.Server/Rachis/Leader.cs

```diff
@@ -329,6 +331,8 @@ private void Run()
                             }
                             _running.Lower();
                             return;
+                        case 4: // an error occurred during EmptyQueue()
+                            throw _errorOccurred.Task.Result;
```

Comment:
We are loosing the stack here. Can just do `_errorOccurred.Task.Wait()` and set this as `SetException`, instead of `SetResult`

Reply:
done